### ISSUE TITLE
Support for php-esque date formatting and parsing

### ISSUE THREAD

**Issue by thomas-p-wilson** [jtwig/jtwig#287]
Presently, given that we use Joda for most of our date/time management, we use Joda-style formatting. We should implement php-esque (ala strtotime) date formatting as outlined in the following documents:

- http://php.net/manual/en/function.date.php (output formatting)
- http://php.net/manual/en/datetime.formats.time.php (time parsing)
- http://php.net/manual/en/datetime.formats.date.php (date parsing)
- http://php.net/manual/en/datetime.formats.compound.php (compound format parsing)
- http://php.net/manual/en/datetime.formats.relative.php (relative date parsing)

**Comment by thomas-p-wilson** [comment]
I have already begun work on this as a by-product of my work with extensions. Will document future issues specific to date parsing/formatting here.

**Comment by realcnbs** [comment]
Are you sure you should mimic strtotime functionality here? Or did you find a library doing that?
Please read my comments on this at #251

**Comment by thomas-p-wilson** [comment]
I did look around for libraries to handle strtotime-style parsing and output, to no avail. Fortunately, it's not at all difficult to implement it using Joda's `DateTimeFormatterBuilder`. Check out what I've done for output formatting so far: https://github.com/thomas-p-wilson/jtwig/blob/feature-extensions/jtwig-core/src/main/java/org/jtwig/util/DateFormatUtil.java. And pardon the lack of implementation of a couple items.

**Comment by thomas-p-wilson** [comment]
If I had remembered #251 earlier I may have left this alone for a bit. Still, I would have done it eventually :smile:

**Comment by thomas-p-wilson** [comment]
FYI, my branch doesn't compile right now. Working out the kinks.

**Comment by realcnbs** [comment]
Looks cool, would be great if you can do it to be 100% compatible. But i think you should externalize that outside of jtwig. So other guys needing this functionality could easily find and use it.

**Comment by thomas-p-wilson** [comment]
I'm definitely thinking I'll externalize it. Once I'm done it will be 100% compatible, but I've only been an hour or so on it so far. @lyncodev, any issues if I pull this abstraction out into a personal project?

**Comment by thomas-p-wilson** [comment]
Well, 100% barring bugs and whatnot of course, but that's what ruthless testing is for no?

**Comment by realcnbs** [comment]
Yeah, maybe look up php sources for tests if they do publish any.

**Comment by thomas-p-wilson** [comment]
They're on Github somewhere. Found them a little while back when I was doing some other Jtwig-related stuff.
